blog/2024-08-08-quorum-queues-in-4.0/index.md

```diff
@@ -0,0 +1,197 @@
+will switch over to the higher priority consumer: it will stop delivering messages to the current
+consumer, wait for all the messages to be acknowledged, and then will deactivate the old consumer,
+and activate the higher priority consumer instead.
+
+The graph below shows this behavior. There are three metrics on this graph:
+* the green line shows the number of messages consumed by the first (default priority) consumer (which happens to be configured to consume 10 msgs/s)
+* yellow, shows the same value but for the second, higher priority consumer
+* blue, shows the number of unacknowledged messages (axis scale on the right)
+
+![Single Active Consumer switchover: the normal-priority consumer (green) gets deactivated after it has acknowledged its messages, then the higher-priority consumer (yellow) gets activated](sac-and-consumer-priority.png)
+
+Initially, we only have one consumer and as expected, it consumes 9-10 msgs/s (these jumps between 9 and 10
+are simply a result of how the metrics are emitted and then displayed). This consumer is configured with the prefetch
+of 1000 messages and since there were many messages in the queue - the prefetch buffer is used to the maximum.
+Then the yellow line appears, initially at 0 msgs/s. This is the higher priority consumer, which is already connected,
+but not yet active. From the moment it connected, we can see the number of unacknowledged messages going down, since the
+queue no longer delivers messages to the original consumer. Once all of them are acknowledged, the new consumer
+becomes the single active consumer and receives 1000 messages, since that's its prefetch value. It then happily
+consumes around 10 msgs/s as configured.
+
+<details>
+    <summary>How Was This Test Performed?</summary>
+
+    For this test we used [perf-test](https://perftest.rabbitmq.com/), a testing client for AMQP 0.9.1.
+    ```
+    # Publish 5000 messages to have a backlog (perf-test will declare a quorum queue `qq-sac`)
+    perf-test --quorum-queue --queue qq-sac --pmessages 5000 --confirm 100 -qa x-single-active-consumer=true --consumers 0
+    # Start a consumer with the default priority and prefetch of 1000; consume ~10 msgs/s
+    perf-test --producers 0 --predeclared --queue qq-sac --consumer-latency 100000 --qos 1000
+    # In another window, some time after starting the first consumer, start a higher priority consumer
+    perf-test --producers 0 --predeclared --queue qq-sac --consumer-latency 100000 --qos 1000 --consumer-args x-priority=10
+    ```
+    After some time you should see that the first consumer stopped receiving messages (no more output from `perf-test`),
+    while the second one receives messages.
+</details>
+
+:::note
+The settings used in this example were chosen to highlight the switchover process and are not great for
+real world scenarios. If a consumer can only process 10 msgs/s, there's usually no reason to configure
+the prefetch value as high as 1000.
+:::
+
+## Checkpoints
```

Comment:
I'd rename this to "Fast Quorum Queue Recovery with Checkpoints" or something like that.